README.md
# tbib_style

A Package for flutter.

You can use the package to control your font easily.

you can control your font with google font package

# tutorial
* [`Youtube`]( https://youtu.be/P1eAeWMCgxM)

# default style should use in desktop

property | description
-------- |------------
h1       | fontSize: 40, fontWeight: FontWeight.bold, height: 1.2
h2       | fontSize: 32, fontWeight: FontWeight.w600, height: 1.2
h3       | fontSize: 28, fontWeight: FontWeight.w600, height: 1.2
h4       | fontSize: 24, fontWeight: FontWeight.w500, height: 1.2
h5       | fontSize: 20, height: 1.2
h6       | fontSize: 16, height: 1.2
b1       | fontSize: 20, fontWeight: FontWeight.w500, height: 1.2
b2       | fontSize: 16, fontWeight: FontWeight.w400, height: 1.2



# important information

in this version can use lisenGoogleFont without choose h number

 TBIBFontStyle.lisenGoogleFont(GoogleFonts.roboto()); // solved in v 1.0.1

when use TBIBFontStyle.lisenGoogleFont(
      GoogleFonts.aBeeZee(textStyle: TBIBFontStyle.h2), 2);

      TBIBFontStyle.addCustomFont("BerthaMelanie", 3);
      
      you see ,2 ?? 2 meaning h2 in this library you have h1 to h6 only and b1 take style from h5 and b2 take style from h6

when use Pacakge google font use it in first code then edit your style

# you can use code in main to handle all fonts in light and dark mode

```dart
  
//h1 style that mean h1 in library change fontFamily with BerthaMelanie
  TBIBFontStyle.h1 = TBIBFontStyle.h1.copyWith(fontFamily: "BerthaMelanie");
  //dark h1
    // 0.0.6 not need change color to white it is default

  TBIBFontStyle.h1Dark = TBIBFontStyle.h1Dark.copyWith(color: Colors.white);

//h2 dark with default style but in dark mode with color white
  // 0.0.6 not need change color to white it is default

  TBIBFontStyle.h2Dark = TBIBFontStyle.h2Dark.copyWith(color: Colors.white);

  //h3 and h3Dark style will change fontFamily with BerthaMelanie like in h1
  // ,3 is mean h3
  TBIBFontStyle.addCustomFont("BerthaMelanie", 3);
  // h3 dark  will be color white 
  // 0.0.6 not need change color to white it is default
  TBIBFontStyle.h3Dark = TBIBFontStyle.h3Dark.copyWith(color: Colors.white);

```
# You can use responsive font size  in 0.0.6 or above

```dart
  double width = MediaQuery.of(context).size.width;
     TBIBFontStyle.responsiveFontSize(width);
     // or use that and make h1 and h1Dark in screen < 1200
     TBIBFontStyle.responsiveFontSize(width ,h1Size: 20,);
     //or use that to make h1 in screen >= 1200 50 px and < 1200 will be 20
     TBIBFontStyle.responsiveFontSize(width ,h1Size: 20, h1SizePc: 50);

```

# when use responsive font size in screen < 1200  default

property | description
-------- |------------
h1       | fontSize: 22 + ((1.5 * screenWidth) / 100), fontWeight: FontWeight.bold, height: 1.2
h2       | fontSize: 21.2 + ((.9 * screenWidth) / 100), fontWeight: FontWeight.w600, height: 1.2
h3       | fontSize: 20.8 + ((.6 * screenWidth) / 100), fontWeight: FontWeight.w600, height: 1.2
h4       | fontSize: 20.4 + ((.3 * screenWidth) / 100), fontWeight: FontWeight.w500, height: 1.2
h5       | fontSize: 20, height: 1.2
h6       | fontSize: 16, height: 1.2
b1       | fontSize: 20, fontWeight: FontWeight.w500, height: 1.2
b2       | fontSize: 16, fontWeight: FontWeight.w400, height: 1.2


# you can use GoogleFont flutter package like this 
  
```dart
      /*
      */
// in all h number font family is roboto from GoogleFont Package 

       TBIBFontStyle.lisenGoogleFont(GoogleFonts.roboto());


         that mean h2 font family is aBeeZee from GoogleFont Package 

    TBIBFontStyle.lisenGoogleFont(
      GoogleFonts.aBeeZee(textStyle: TBIBFontStyle.h2), 2);
```

# you can use custom font 
 
```dart
    /*
    I download BerthaMelanie font and added it in pubspec 
    and when add this code in main it mean h3 and h3Dark fontFamily = BerthaMelanie
    */
   TBIBFontStyle.addCustomFont("BerthaMelanie", 3);
```

# you can use deafult style in flutter h1 = headline1 but should remove responsive font size

```dart
   TBIBFontStyle.defaultFlutterStyle();
```

# when use light and dark mode

```dart
    Text(
        "Text color mode",
        style: isDark ? TBIBFontStyle.h3Dark : TBIBFontStyle.h3,
    ),
```
## now use ResponsFont class with flutter_screenutil like that 


## this is deprecated and not working properly 
# You can use responsive font size  TBIBFontStyle.fontResponsive(width ); deprecated

```dart
        MaterialApp(
        debugShowCheckedModeBanner: false,
        title: 'Choose Two Squares',
        builder: (context, child) {
            ScreenUtil.setContext(context);

            ResponiveFont.responsive();
            BotToastInit();

            child = child; //do something
            child = botToastBuilder(context, child);
            return child;
        },
            ...
```

## this is deprecated and not working properly 
# You can use responsive font size  TBIBFontStyle.fontResponsive(width ); deprecated

```dart
    /*
    that mean use font responsive with default style library in mobile 
    */
    double width = MediaQuery.of(context).size.width;
    TBIBFontStyle.fontResponsive(width );
```

# You can use responsive font size deprecated

```dart
    /*
    that mean use font responsive with default style library in mobile 
    */
    double width = MediaQuery.of(context).size.width;
   
```
 
```dart
/*
 that code mean when width screen > 1200 h1 will be font size 20
*/
 double width = MediaQuery.of(context).size.width;
   if(width < 1200){
       TBIBFontStyle.h1=TBIBFontStyle.h1.copyWith(fontSize:20); 
       // or remove default style
       TBIBFontStyle.h1 = TextStyle(fontSize: 20);
         // when use useCastomFontSize is mean use any h1 to h6 changed
       TBIBFontStyle.fontResponsive(width,useCastomFontSize: true); // deprecated
    }

if make useCastomFontSize true you should use hotRestart not reload sorry for this issues


CHANGELOG.md
## 2.0.0
 now can use class ResponiveFont and  use it with flutter_screenutil
## 1.0.1
* now solve font style default removed when use google font
## 1.0.0
* solve b2 color in dark mode 
  and now you can handle all h with google font
## 0.0.6 - 0.0.7 - 0.0.8

* Solve font responsive issues with new function and make dark font color white default


## 0.0.5

* add video and solve some problems -- and change example code.

## 0.0.3 and ## 0.0.4

* readme dart code with comment.


## 0.0.2

* add Github url.


## 0.0.1

*  Describe initial release.



example/lib/main.dart
import 'dart:developer';

import 'package:flutter/material.dart';
import 'package:google_fonts/google_fonts.dart';
import 'package:tbib_style/style/font_style.dart';

void main() {
  // use font google in h1 and h1Dark
  TBIBFontStyle.lisenGoogleFont(
    GoogleFonts.aBeeZee(),
  );

  log(TBIBFontStyle.h2.fontSize.toString());

  // use custom font in h1 only

  TBIBFontStyle.h1 = TBIBFontStyle.h1.copyWith(fontFamily: "BerthaMelanie");

  // use custom font in h3 and h3Dark only
  TBIBFontStyle.addCustomFont("BerthaMelanie", 3);

  runApp(const App());
}

class App extends StatelessWidget {
  const App({Key? key}) : super(key: key);

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(
        title: "FontStyle Example",
      ),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  bool isDark = false;
  @override
  Widget build(BuildContext context) {
    double width = MediaQuery.of(context).size.width;
// responsive font in screen > 1200 will be font size like bootstrap
// in < 1200 will be calc font like bootstrap
    TBIBFontStyle.responsiveFontSize(width);

    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
        actions: [
          IconButton(
              onPressed: () {
                setState(() {
                  isDark = !isDark;
                });
              },
              icon: const Icon(Icons.brightness_4))
        ],
      ),
      body: Container(
          color: isDark ? Colors.black : null,
          child: Center(
            child: Column(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                Text(
                  "Text custom font",
                  style: isDark ? TBIBFontStyle.h1Dark : TBIBFontStyle.h1,
                ),
                const SizedBox(
                  height: 20,
                ),
                Text(
                  "Text color mode",
                  style: isDark ? TBIBFontStyle.h2Dark : TBIBFontStyle.h2,
                ),
                const SizedBox(
                  height: 20,
                ),
                Text(
                  "Text color mode",
                  style: isDark ? TBIBFontStyle.h3Dark : TBIBFontStyle.h3,
                ),
              ],
            ),
          )),
    );
  }
}


lib/responsive_font.dart
import 'package:flutter_screenutil/flutter_screenutil.dart';

import 'tbib_style.dart';

class ResponiveFont {
  static bool _responsiveEnabled = false;

  static void responsive() {
    if (!_responsiveEnabled) {
      TBIBFontStyle.h1 =
          TBIBFontStyle.h1.copyWith(fontSize: TBIBFontStyle.h1.fontSize!.sp);
      TBIBFontStyle.h1Dark = TBIBFontStyle.h1Dark
          .copyWith(fontSize: TBIBFontStyle.h1Dark.fontSize!.sp);

      TBIBFontStyle.h2 =
          TBIBFontStyle.h2.copyWith(fontSize: TBIBFontStyle.h2.fontSize!.sp);
      TBIBFontStyle.h2Dark = TBIBFontStyle.h2Dark
          .copyWith(fontSize: TBIBFontStyle.h2Dark.fontSize!.sp);

      TBIBFontStyle.h3 =
          TBIBFontStyle.h3.copyWith(fontSize: TBIBFontStyle.h3.fontSize!.sp);
      TBIBFontStyle.h3Dark = TBIBFontStyle.h3Dark
          .copyWith(fontSize: TBIBFontStyle.h3Dark.fontSize!.sp);

      TBIBFontStyle.h4 =
          TBIBFontStyle.h4.copyWith(fontSize: TBIBFontStyle.h4.fontSize!.sp);
      TBIBFontStyle.h4Dark = TBIBFontStyle.h4Dark
          .copyWith(fontSize: TBIBFontStyle.h4Dark.fontSize!.sp);

      TBIBFontStyle.h5 =
          TBIBFontStyle.h5.copyWith(fontSize: TBIBFontStyle.h5.fontSize!.sp);
      TBIBFontStyle.h5Dark = TBIBFontStyle.h5Dark
          .copyWith(fontSize: TBIBFontStyle.h5Dark.fontSize!.sp);

      TBIBFontStyle.h6 =
          TBIBFontStyle.h6.copyWith(fontSize: TBIBFontStyle.h6.fontSize!.sp);
      TBIBFontStyle.h6Dark = TBIBFontStyle.h6Dark
          .copyWith(fontSize: TBIBFontStyle.h6Dark.fontSize!.sp);

      TBIBFontStyle.b1 =
          TBIBFontStyle.b1.copyWith(fontSize: TBIBFontStyle.b1.fontSize!.sp);
      TBIBFontStyle.b1Dark = TBIBFontStyle.b1Dark
          .copyWith(fontSize: TBIBFontStyle.b1Dark.fontSize!.sp);

      TBIBFontStyle.b2 =
          TBIBFontStyle.b2.copyWith(fontSize: TBIBFontStyle.b2.fontSize!.sp);
      TBIBFontStyle.b2Dark = TBIBFontStyle.b2Dark
          .copyWith(fontSize: TBIBFontStyle.b2Dark.fontSize!.sp);
      _responsiveEnabled = true;
    }
  }
}


lib/style/font_style.dart
import 'package:flutter/material.dart';

class TBIBFontStyle {
  static TextStyle h1 = TextStyle(
      fontSize: 40,
      fontWeight: FontWeight.bold,
      height: 1.2,
      color: Colors.black);
  static TextStyle h2 = TextStyle(
      fontSize: 32,
      fontWeight: FontWeight.w600,
      height: 1.2,
      color: Colors.black);
  static TextStyle h3 = TextStyle(
      fontSize: 28,
      fontWeight: FontWeight.w600,
      height: 1.2,
      color: Colors.black);
  static TextStyle h4 = TextStyle(
      fontSize: 24,
      fontWeight: FontWeight.w500,
      height: 1.2,
      color: Colors.black);
  static TextStyle h5 =
      TextStyle(fontSize: 20, height: 1.2, color: Colors.black);
  static TextStyle h6 =
      TextStyle(fontSize: 16, height: 1.2, color: Colors.black);
  static TextStyle b1 = TextStyle(
      fontSize: 20,
      fontWeight: FontWeight.w500,
      height: 1.2,
      color: Colors.black);
  static TextStyle b2 = TextStyle(
      fontSize: 16,
      fontWeight: FontWeight.w400,
      height: 1.2,
      color: Colors.black);

  static TextStyle h1Dark = TextStyle(
      fontSize: 40,
      fontWeight: FontWeight.bold,
      height: 1.2,
      color: Colors.white);
  static TextStyle h2Dark = TextStyle(
      fontSize: 32,
      fontWeight: FontWeight.w600,
      height: 1.2,
      color: Colors.white);
  static TextStyle h3Dark = TextStyle(
      fontSize: 28,
      fontWeight: FontWeight.w600,
      height: 1.2,
      color: Colors.white);
  static TextStyle h4Dark = TextStyle(
      fontSize: 24,
      fontWeight: FontWeight.w500,
      height: 1.2,
      color: Colors.white);
  static TextStyle h5Dark =
      TextStyle(fontSize: 20, height: 1.2, color: Colors.white);
  static TextStyle h6Dark =
      TextStyle(fontSize: 16, height: 1.2, color: Colors.white);
  static TextStyle b1Dark = TextStyle(
      fontSize: 20,
      fontWeight: FontWeight.w500,
      height: 1.2,
      color: Colors.white);
  static TextStyle b2Dark = TextStyle(
      fontSize: 16,
      fontWeight: FontWeight.w400,
      height: 1.2,
      color: Colors.white);

  static defaultFlutterStyle() {
    h1 = TextStyle(
      fontSize: 96,
      fontWeight: FontWeight.w200,
    );
    h2 = TextStyle(
      fontSize: 60,
      fontWeight: FontWeight.w200,
    );
    h3 = TextStyle(fontSize: 48, fontWeight: FontWeight.w300);
    h4 = TextStyle(
      fontSize: 34,
      fontWeight: FontWeight.w300,
    );
    h5 = TextStyle(fontSize: 24, fontWeight: FontWeight.w300);
    h6 = TextStyle(
      fontSize: 20,
      fontWeight: FontWeight.w500,
    );
    b1 = TextStyle(fontSize: 34, fontWeight: FontWeight.w500);
    b2 = TextStyle(fontSize: 20, fontWeight: FontWeight.w400);
  }

  static lisenGoogleFont(TextStyle googleFontsName, {int? h}) {
    if (h != null) {
      switch (h) {
        case 1:
          {
            h1Dark = h1 = googleFontsName;
            h1Dark = h1Dark.copyWith(color: Colors.white);
            return;
          }
        case 2:
          {
            h2Dark = h2 = googleFontsName;
            h2Dark = h2Dark.copyWith(color: Colors.white);
            return;
          }
        case 3:
          {
            h3Dark = h3 = googleFontsName;
            h3Dark = h3Dark.copyWith(color: Colors.white);

            return;
          }
        case 4:
          {
            h4Dark = h4 = googleFontsName;
            h4Dark = h4Dark.copyWith(color: Colors.white);

            return;
          }
        case 5:
          {
            h5Dark = h5 = b1 = b1Dark = googleFontsName;
            h5Dark = h5Dark.copyWith(color: Colors.white);
            b1Dark = b1Dark.copyWith(color: Colors.white);

            return;
          }
        case 6:
          {
            h6 = h6Dark = b2 = b2Dark = googleFontsName;
            h6Dark = h6Dark.copyWith(color: Colors.white);
            b2Dark = b2Dark.copyWith(color: Colors.white);

            return;
          }
        default:
          {
            return null;
          }
      }
    } else {
      h1Dark = h1 = googleFontsName.copyWith(
          fontSize: 40, fontWeight: FontWeight.bold, height: 1.2);
      h1Dark = h1Dark.copyWith(color: Colors.white);

      h2Dark = h2 = googleFontsName.copyWith(
          fontSize: 32, fontWeight: FontWeight.w600, height: 1.2);
      h2Dark = h2Dark.copyWith(color: Colors.white);

      h3Dark = h3 = googleFontsName.copyWith(
          fontSize: 28, fontWeight: FontWeight.w600, height: 1.2);
      h3Dark = h3Dark.copyWith(color: Colors.white);

      h4Dark = h4 = googleFontsName.copyWith(
          fontSize: 24, fontWeight: FontWeight.w500, height: 1.2);
      h4Dark = h4Dark.copyWith(color: Colors.white);

      h5Dark = h5 =
          b1 = b1Dark = googleFontsName.copyWith(fontSize: 20, height: 1.2);
      h5Dark = h5Dark.copyWith(color: Colors.white);
      b1Dark = b1Dark.copyWith(color: Colors.white);

      h6 = h6Dark =
          b2 = b2Dark = googleFontsName.copyWith(fontSize: 16, height: 1.2);
      h6Dark = h6Dark.copyWith(color: Colors.white);
      b2Dark = b2Dark.copyWith(color: Colors.white);
    }
  }

  @Deprecated("Use responsiveFontSize instead. it is deprecated in 0.0.6")
  static void fontResponsive(double screenWidth,
      {bool useCastomFontSize = false}) {
    if (!useCastomFontSize) {
      if (screenWidth < 1200) {
        h1 = h1.copyWith(fontSize: 22 + ((1.5 * screenWidth) / 100));
        h1Dark = h1Dark.copyWith(fontSize: 22 + ((1.5 * screenWidth) / 100));

        h2 = h2.copyWith(fontSize: 21.2 + ((.9 * screenWidth) / 100));
        h2Dark = h2Dark.copyWith(fontSize: 21.2 + ((.9 * screenWidth) / 100));

        h3 = h3.copyWith(fontSize: 20.8 + ((.6 * screenWidth) / 100));
        h3Dark = h3Dark.copyWith(fontSize: 20.8 + ((.6 * screenWidth) / 100));
        h4 = h4.copyWith(fontSize: 20.4 + ((.3 * screenWidth) / 100));
        h4Dark = h4Dark.copyWith(fontSize: 20.4 + ((.3 * screenWidth) / 100));
      } else {
        h1 =
            h1.copyWith(fontSize: 40, fontWeight: FontWeight.bold, height: 1.2);
        h1Dark = h1Dark.copyWith(
            fontSize: 40, fontWeight: FontWeight.bold, height: 1.2);
        h2 =
            h2.copyWith(fontSize: 32, fontWeight: FontWeight.w600, height: 1.2);
        h2Dark = h2Dark.copyWith(
            fontSize: 32, fontWeight: FontWeight.w600, height: 1.2);
        h3 =
            h3.copyWith(fontSize: 28, fontWeight: FontWeight.w600, height: 1.2);
        h3Dark = h3Dark.copyWith(
            fontSize: 28, fontWeight: FontWeight.w600, height: 1.2);
        h4 = h4.copyWith(fontSize: 24, fontWeight: FontWeight.w500);
        h4Dark = h4Dark.copyWith(fontSize: 24, fontWeight: FontWeight.w500);
      }
    } else {
      if (screenWidth < 1200) {
        h1 = h1.copyWith(fontSize: h1.fontSize! + ((1.5 * screenWidth) / 100));
        h1Dark = h1Dark.copyWith(
            fontSize: h1.fontSize! + ((1.5 * screenWidth) / 100));

        h2 = h2.copyWith(fontSize: h2.fontSize! + ((.9 * screenWidth) / 100));
        h2Dark = h2Dark.copyWith(
            fontSize: h2.fontSize! + ((.9 * screenWidth) / 100));

        h3 = h3.copyWith(fontSize: h3.fontSize! + ((.6 * screenWidth) / 100));
        h3Dark = h3Dark.copyWith(
            fontSize: h3.fontSize! + ((.6 * screenWidth) / 100));
        h4 = h4.copyWith(fontSize: h4.fontSize! + ((.3 * screenWidth) / 100));
        h4Dark = h4Dark.copyWith(
            fontSize: h4.fontSize! + ((.3 * screenWidth) / 100));
      } else {
        h1 = h1.copyWith(
            fontSize: h1.fontSize!, fontWeight: FontWeight.bold, height: 1.2);
        h1Dark = h1Dark.copyWith(
            fontSize: h1.fontSize!, fontWeight: FontWeight.bold, height: 1.2);
        h2 = h2.copyWith(
            fontSize: h2.fontSize!, fontWeight: FontWeight.w600, height: 1.2);
        h2Dark = h2Dark.copyWith(
            fontSize: h2.fontSize!, fontWeight: FontWeight.w600, height: 1.2);
        h3 = h3.copyWith(
            fontSize: h3.fontSize!, fontWeight: FontWeight.w600, height: 1.2);
        h3Dark = h3Dark.copyWith(
            fontSize: h3.fontSize!, fontWeight: FontWeight.w600, height: 1.2);
        h4 = h4.copyWith(fontSize: h4.fontSize!, fontWeight: FontWeight.w500);
        h4Dark = h4Dark.copyWith(
            fontSize: h4.fontSize!, fontWeight: FontWeight.w500);
      }
    }
  }

  static void responsiveFontSize(double screenWidth,
      {double h1Size = 22,
      double h1SizePc = 40,
      double h2Size = 21.2,
      double h2SizePc = 32,
      double h3Size = 20.8,
      double h3SizePc = 28,
      double h4Size = 20.4,
      double h4SizePc = 24}) {
    if (screenWidth < 1200) {
      h1 = h1.copyWith(fontSize: h1Size - ((.9 * screenWidth) / 100));
      h1Dark = h1Dark.copyWith(fontSize: h1Size - ((.9 * screenWidth) / 100));

      h2 = h2.copyWith(fontSize: h2Size - ((.9 * screenWidth) / 100));
      h2Dark = h2Dark.copyWith(fontSize: h2Size - ((.9 * screenWidth) / 100));

      b1 = h3 = h3.copyWith(fontSize: h3Size - ((.9 * screenWidth) / 100));
      b1Dark = h3Dark =
          h3Dark.copyWith(fontSize: h3Size - ((.9 * screenWidth) / 100));
      b2 = h4 = h4.copyWith(fontSize: h4Size - ((.9 * screenWidth) / 100));
      b2Dark = h4Dark =
          h4Dark.copyWith(fontSize: h4Size - ((.9 * screenWidth) / 100));
    } else {
      h1 = h1.copyWith(fontSize: h1SizePc + ((1.5 * screenWidth) / 100));
      h1Dark =
          h1Dark.copyWith(fontSize: h1SizePc + ((1.5 * screenWidth) / 100));

      h2 = h2.copyWith(fontSize: h2SizePc + ((.9 * screenWidth) / 100));
      h2Dark = h2Dark.copyWith(fontSize: h2SizePc + ((.9 * screenWidth) / 100));

      b1 = h3 = h3.copyWith(fontSize: h3SizePc + ((.6 * screenWidth) / 100));
      b1Dark = h3Dark =
          h3Dark.copyWith(fontSize: h3SizePc + ((.6 * screenWidth) / 100));
      b2 = h4 = h4.copyWith(fontSize: h4SizePc + ((.3 * screenWidth) / 100));
      b2Dark = h4Dark =
          h4Dark.copyWith(fontSize: h4SizePc + ((.3 * screenWidth) / 100));
    }
  }

  static void addCustomFont(String yourFontFamily, int? h) {
    if (h != null) {
      switch (h) {
        case 1:
          {
            h1 = h1.copyWith(fontFamily: yourFontFamily);
            h1Dark = h1Dark.copyWith(fontFamily: yourFontFamily);

            break;
          }
        case 2:
          {
            h2 = h2.copyWith(fontFamily: yourFontFamily);
            h2Dark = h2Dark.copyWith(fontFamily: yourFontFamily);
            break;
          }
        case 3:
          {
            h3 = h3.copyWith(fontFamily: yourFontFamily);
            h3Dark = h3Dark.copyWith(fontFamily: yourFontFamily);
            break;
          }
        case 4:
          {
            h4 = h4.copyWith(fontFamily: yourFontFamily);
            h4Dark = h4Dark.copyWith(fontFamily: yourFontFamily);
            break;
          }
        case 5:
          {
            h5 = h5.copyWith(fontFamily: yourFontFamily);
            h5Dark = h5Dark.copyWith(fontFamily: yourFontFamily);
            b1 = b1.copyWith(fontFamily: yourFontFamily);
            b1Dark = b1Dark.copyWith(fontFamily: yourFontFamily);
            break;
          }
        case 6:
          {
            h6 = h6.copyWith(fontFamily: yourFontFamily);
            h6Dark = h6Dark.copyWith(fontFamily: yourFontFamily);
            b2 = b2.copyWith(fontFamily: yourFontFamily);
            b2Dark = b2Dark.copyWith(fontFamily: yourFontFamily);
            break;
          }
        default:
          {
            break;
          }
      }
    } else {
      h1 = h1.copyWith(fontFamily: yourFontFamily);
      h1Dark = h1Dark.copyWith(fontFamily: yourFontFamily);
      h2 = h2.copyWith(fontFamily: yourFontFamily);
      h2Dark = h2Dark.copyWith(fontFamily: yourFontFamily);
      h3 = h3.copyWith(fontFamily: yourFontFamily);
      h3Dark = h3Dark.copyWith(fontFamily: yourFontFamily);
      h4 = h4.copyWith(fontFamily: yourFontFamily);
      h4Dark = h4Dark.copyWith(fontFamily: yourFontFamily);
      h5 = h5.copyWith(fontFamily: yourFontFamily);
      h5Dark = h5Dark.copyWith(fontFamily: yourFontFamily);
      h6 = h6.copyWith(fontFamily: yourFontFamily);
      h6Dark = h6Dark.copyWith(fontFamily: yourFontFamily);
      b1 = b1.copyWith(fontFamily: yourFontFamily);
      b1Dark = b1Dark.copyWith(fontFamily: yourFontFamily);
      b2 = b2.copyWith(fontFamily: yourFontFamily);
      b2Dark = b2Dark.copyWith(fontFamily: yourFontFamily);
    }
  }
}
